src/opnsense/scripts/openvpn/user_pass_verify.php

```diff
@@ -80,16 +80,15 @@ function do_auth($common_name, $serverid, $method, $auth_file)
     if (empty($username) || empty($password)) {
         return "username or password missing ({$method} - {$auth_file})";
     }
+    $pinpassword = false;
+    $pin = false;
     if (strpos($password, 'SCRV1:') === 0) {
         // static-challenge https://github.com/OpenVPN/openvpn/blob/v2.4.7/doc/management-notes.txt#L1146
-        // validate and concat password into our default pin+password
+        // validate and split password into pin and password (used below based on authenticato settings)
         $tmp = explode(':', $password);
         if (count($tmp) == 3) {
-            $pass = base64_decode($tmp[1]);
+            $pinpassword = base64_decode($tmp[1]);
```

Comment:
all of this just obscures the intention...

Reply:
We might move this also inside the authenticator loop for clarity, just it didn't seem programmatically correct to me.